### ISSUE TITLE
backend: add ability to configure extensions

### ISSUE THREAD

**Issue by pawlh** [softwareconstruction240/autograder#32]

**Comment by frozenfrank** [comment]
While talking with @DallinFromEarth, we identified that a good route to solve this problem would be to always read the due date out of Canvas which can be configured on a per-student level. This way instructors and TA's can continue managing the deadlines as they always have, and the autograder will handle it correctly.

**Comment by pawlh** [comment]
I think this would be a reasonable solution, but the professors have
decided previously that they do not want to update due dates in Canvas (I'm
happy to give more context for their reasoning). Despite that, we can bring
this up again to see if they'd like to review that stance.

**Comment by frozenfrank** [comment]
@pawlh I didn't realize they cared _where_ they are updated the due dates. Do they want to be managing the due dates directly in the app, or is there another way that would work better for them?

**Comment by frozenfrank** [comment]
Per our discussion in the TA meeting, we decided that extensions will be managed solely in Canvas as exceptions per student. 

A concern was raised about handfuls of students with accommodations letters. In these cases, they are allowed several days extra for _every assignment_ which requires manual effort to input into Canvas. We decided that the TAs will take the burden of this effort from the instructors to achieve two main goals:
1. Make Canvas a canonical source of truth for students to see their specific due date
2. Simplify the development of the autograder functionality

We also discussed blending these approaches by implementing some data storage and understanding of students with accommodations. This would undermine both of the objectives above, but it would come with the following pro:
1. Manual effort may be saved from the process of inputting the exceptions to each assignment (which I understand have to be done on a per-student, per-assignment basis).

That said, there may be a better alternative. Namely, we may be able to write a script to programmatically add these due date exceptions. This would meet all the above goals, and also eliminate the manual effort required. 

Therefore, I recommend closing this issue and creating the following issue: **support: create tool to batch create extensions for students with accommodations**. Work on #42 will move forward independent of this request.
